cmd/cstor-pool-mgmt/cstorops/pool/pool.go

```diff
@@ -0,0 +1,150 @@
+
+package pool
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+)
+
+// PoolOperator is the name of the tool that makes pool-related operations.
+const (
+	PoolOperator = "zpool"
+)
+
+// ImportPool imports cStor pool if already present.
+func ImportPool(cStorPoolUpdated *apis.CStorPool) error {
+	cmdimport := importPoolBuilder(cStorPoolUpdated)
+	stdoutStderrImport, err := cmdimport.CombinedOutput()
+	if err != nil {
+		glog.Error("Pool import err: ", err)
+		glog.Error("stdoutStderr: ", string(stdoutStderrImport))
+		return err
+	}
+
+	glog.Info("Importing Pool Successful")
+	return nil
+}
+
+// importPoolBuilder is to build pool import command.
+func importPoolBuilder(cStorPoolUpdated *apis.CStorPool) *exec.Cmd {
+	// populate pool import attributes.
+	var importAttr []string
+	importAttr = append(importAttr, "import")
+	if cStorPoolUpdated.Spec.PoolSpec.CacheFile != "" {
+		importAttr = append(importAttr, "-c", cStorPoolUpdated.Spec.PoolSpec.CacheFile)
+	}
+
+	importAttr = append(importAttr, cStorPoolUpdated.Spec.PoolSpec.PoolName)
+
+	// execute import pool command.
+	cmdimport := exec.Command(PoolOperator, importAttr...)
+	return cmdimport
+}
+
+// CreatePool creates a new cStor pool.
+func CreatePool(cStorPoolUpdated *apis.CStorPool) error {
+
+	poolCreateCmd := createPoolBuilder(cStorPoolUpdated)
+
+	glog.V(4).Info("poolCreateCmd : ", poolCreateCmd)
+	stdoutStderr, err := poolCreateCmd.CombinedOutput()
+	if err != nil {
+		glog.Error("err: ", err)
+		glog.Error("stdoutStderr: ", string(stdoutStderr))
+		return err
+	}
+	glog.Info("Creating Pool Successful")
+	return nil
+}
+
+// createPoolBuilder is to build create pool command.
+func createPoolBuilder(cStorPoolUpdated *apis.CStorPool) *exec.Cmd {
+	// populate pool creation attributes.
+	var createAttr []string
+	createAttr = append(createAttr, "create", "-f", "-o")
+	if cStorPoolUpdated.Spec.PoolSpec.CacheFile != "" {
+		cachefile := "cachefile=" + cStorPoolUpdated.Spec.PoolSpec.CacheFile
+		createAttr = append(createAttr, cachefile)
+	}
+
+	createAttr = append(createAttr, cStorPoolUpdated.Spec.PoolSpec.PoolName)
+
+	for _, disk := range cStorPoolUpdated.Spec.Disks.DiskList {
+		createAttr = append(createAttr, disk)
+	}
+
+	//execute pool creation command.
+	poolCreateCmd := exec.Command(PoolOperator, createAttr...)
+	return poolCreateCmd
+}
+
+// CheckValidPool checks for validity of CStorPool resource.
+func CheckValidPool(cStorPoolUpdated *apis.CStorPool) error {
+	if cStorPoolUpdated.Spec.PoolSpec.PoolName == "" {
+		return fmt.Errorf("Poolname cannot be empty")
+	}
+	if len(cStorPoolUpdated.Spec.Disks.DiskList) < 1 {
+		return fmt.Errorf("Disk name(s) cannot be empty")
+	}
+	return nil
+}
+
+// GetPoolName return the pool already created.
+func GetPoolName() (string, error) {
+	poolnameStr := PoolOperator + " status | grep pool:"
+	poolnamecmd := exec.Command("bash", "-c", poolnameStr)
```

Comment:
why are we not using exec.command(pooOperator, ....) like the other places?

Reply:
That is because in other places, we build command from array of strings from struct object. Here we build cmd from string - " status | grep pool:".

Reply:
Ok I see, but why are we using different approaches in the same code (and file even) is it not better to use a common approach for this?

Reply:
faced some issues using | in sliced cmd. Since we are replacing with  'get -Hp name' I'm trying with slice(array)